src/psyclone/dynamo0p3.py

```diff
@@ -1871,15 +1871,15 @@ class DynMeshes(object):
     is "restriction" where a field on a fine mesh is mapped onto a coarse
     mesh.
 
-    :param schedule: the schedule of the Invoke for which to extract \
-                     information on all required inter-grid operations.
-    :type schedule: :py:class:`psyclone.dynamo0p3.DynSchedule`
+    :param invoke: the Invoke for which to extract information on all \
+                   required inter-grid operations.
+    :type invoke: :py:class:`psyclone.dynamo0p3.DynInvoke`
     :param unique_psy_vars: list of arguments to the PSy-layer routine.
     :type unique_psy_vars: list of \
                       :py:class:`psyclone.dynamo0p3.DynKernelArgument` objects.
     '''
 
-    def __init__(self, schedule, unique_psy_vars):
+    def __init__(self, invoke, unique_psy_vars):
```

Comment:
I needed the Invoke object rather than just the Schedule because I needed to call `Invoke.iterate_over_dofs_only` when deciding whether or not to create a Mesh object.

Reply:
Sure, but you could have done schedule.invoke. I guess the point is that it makes sense to pass in the outer class (in the PSy hierarchy) rather than referencing back, which I agree with.